File: backend/CertificatesManagerApi/Mappers/CommentMapper.cs
using CertificatesManagerApi.DTOs;
using CertificatesManagerApi.Entities;

namespace CertificatesManagerApi.Mappers
{
    public class CommentMapper
    {
        public static CommentDTO ToDto(Comment comment)
        {
            CommentDTO commentDto = new()
            {
                UserName = comment.User.Name,
                Comment = comment.CommentText
            };

            return commentDto;
        }

        public static Comment ToEntity(CreateCommentDTO createCommentDTO, Certificate commentCertificate, User commentUser)
        {
            Comment comment = new()
            {
                CommentText = createCommentDTO.CommentText,
                Certificate = commentCertificate,
                User = commentUser
            };

            return comment;
        }
    }
}

File: frontend/src/common/utils/mappers/comment-mapper.ts
import UserComment from '../../models/comment.model';
import { CommentDTO, CommentPostDTO } from '../../models/dtos/comment-dto';

export class CommentMapper {
	static ToModel(commentDTO: CommentDTO): UserComment {
		const comment: UserComment = {
			userName: commentDTO.userName,
			comment: commentDTO.comment,
		};

		return comment;
	}

	static ToDTO(comment: UserComment, userId: string): CommentPostDTO {
		const commentPostDTO: CommentPostDTO = {
			userHandle: userId,
			commentText: comment.comment,
		};

		return commentPostDTO;
	}
}
